src/engine/MaxConstraint.cpp

```diff
@@ -481,22 +613,38 @@ void MaxConstraint::updateVariableIndex( unsigned oldIndex, unsigned newIndex )
         _f = newIndex;
 
     if ( _elements.exists( oldIndex ) )
-	{
-	    _elements.erase( oldIndex );
-	    _elements.insert( newIndex );
-	}
+    {
+        _elements.erase( oldIndex );
+        _elements.insert( newIndex );
+    }
 }
 
 bool MaxConstraint::constraintObsolete() const
 {
     return _obsolete;
 }
 
-void MaxConstraint::eliminateVariable( unsigned var, double /*value*/ )
+void MaxConstraint::eliminateVariable( unsigned var, double value )
 {
+    _eliminatedVariables = true;
+    _maxValueOfEliminated = FloatUtils::max( value, _maxValueOfEliminated );
+    _maxLowerBound = FloatUtils::max( _maxLowerBound, _maxValueOfEliminated );
+
     _elements.erase( var );
-    if ( var == _f || getParticipatingVariables().size() == 1 )
+    if ( var == _f )
         _obsolete = true;
+    if ( getParticipatingVariables().size() == 1 )
+    {
+        if ( !_eliminatedVariables )
+            _obsolete = true;
+
+        // no eliminated variables
+        if ( FloatUtils::gte( value, _maxValueOfEliminated ))
```

Comment:
I didn't understand this part. We already know that _maxValue >= value, because we updated it?

Reply:
you are right. fixed it.